src/components/carmel-gearvr-controls.js

```diff
@@ -0,0 +1,200 @@
+var registerComponent = require('../core/component').registerComponent;
+var bind = require('../utils/bind');
+var getGamepadsByPrefix = require('../utils/tracked-controls').getGamepadsByPrefix;
+var THREE = require('../lib/three');
+
+var GAMEPAD_ID_PREFIX = 'Gear VR Touchpad';
+
+/**
+ * Vive Controls Component
+ * Interfaces with vive controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and highlights the pressed buttons
+ */
+module.exports.Component = registerComponent('carmel-gearvr-controls', {
+  // since aframe-teleport-controls looks for tracked-controls, add one
+  dependencies: ['tracked-controls'],
```

Comment:
renamed to `gearvr-controls` and added support for Samsung Internet GearVR.  (The test scene I use looks low-res in that browser for some reason, but the functionality works as expected.)